findatastructures/src/finset/mod.rs

```diff
@@ -0,0 +1,29 @@
+pub trait FinSet: PartialEq + Clone {
```

Comment:
Same as for the map. The constraints should not be needed and the trait needs a description.